test/testleakautovar.cpp

```diff
@@ -1129,6 +1130,22 @@ class TestLeakAutoVar : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void test6() { // #7685
+        check("void do_wordexp() {\n"
```

Comment:
please use a shorter example code. I would say this will reproduce the problem:

    void do_wordexp(FILE *f) {
        free(getword(f)); // commenting this line out causes the FP to disappear
        fclose(f);
    }